File: examples/zig/src/services.zig
// Compute helpers behind the example program.
pub fn fib(n: u32) u64 {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

pub fn mix(n: u32) u64 {
    var sum: u64 = 0;
    var i: u32 = 0;
    while (i < n) : (i += 1) {
        sum +%= fib(i % 12) * 3;
    }
    return sum;
}

File: examples/node/lib/services.js
// Business logic behind the example server's handlers.
const USERS = [
  { id: '1', name: 'Ada Lovelace' },
  { id: '2', name: 'Grace Hopper' },
  { id: '3', name: 'Linus Torvalds' },
];

const counters = { requests: 0, fibCalls: 0 };

function fib(n) {
  counters.fibCalls++;
  if (n < 2) return n;
  return fib(n - 1) + fib(n - 2);
}

function listUsers() {
  counters.requests++;
  return USERS.map((user) => user.id);
}

function getUser(id) {
  counters.requests++;
  return USERS.find((user) => user.id === id) || null;
}

function stats() {
  const memory = process.memoryUsage();
  const active = typeof process.getActiveResourcesInfo === 'function'
    ? process.getActiveResourcesInfo().length
    : 0;
  return {
    requests: counters.requests,
    fibCalls: counters.fibCalls,
    activeResources: active,
    heapUsed: memory.heapUsed,
    rss: memory.rss,
  };
}

module.exports = { fib, listUsers, getUser, stats };
